### ISSUE TITLE
`GRANT ALL ON public.test  TO regress_dep_user2;` this works, but `GRANT ALL ON ALL TABLES IN SCHEMA testns TO regress_dep_user1;` does not work.  We decided that we could just forward it right away to all datanodes and let it work/fail depending on the presence of the schema. Tables that are created distributed on the datanodes will get the privileges without us worrying about which tables to forward the command for.

### ISSUE THREAD

**Issue by erimatnor** [timescale/timescaledb#3595]

**Comment by mkindahl** [comment]
Removing multinode label since this is not specific to multinode, it applies to normal hypertables as well.

**Comment by nikkhils** [comment]
So there are two issues here:

1) In multi-node the query is not shipped to the datanodes. This is being addressed

2) The main hypertable might reside in the specified SCHEMA, but its chunk will reside in the `_timescaledb_internal` SCHEMA. So what needs to happen is that we collect all the hypertables (single-node or distributed) and ensure that the GRANT reaches these chunks as well.

**Comment by nikkhils** [comment]
Yes, now that I look at this closely, supporting `GRANT ALL ON ALL TABLES IN SCHEMA` might be more work. 

For this GrantStmt, the `targtype` field is `ACL_TARGET_ALL_IN_SCHEMA` whereas in regular `GRANT ON TABLE xyz..`, the `targtype` field is `ACL_TARGET_OBJECT`. In the latter case the `objects` list contains a list of relation range vars whereas in the former it is the name of the schema. 

So, we cannot just add the schema/names of the hypertables/chunks in the `ACL_TARGET_ALL_IN_SCHEMA` case to the `objects` list because it's interpreted entirely differently :-|
